specutils/manipulation/utils.py

```diff
@@ -34,56 +100,74 @@ def linear_exciser(spectrum, region):
+    modified_flux = flux
+    if spectrum.uncertainty is not None:
+        new_uncertainty = spectrum.uncertainty.copy()
+    else:
+        new_uncertainty = None
+
+    # Need to add a check that the subregions don't overlap, since that could
+    # cause undesired results. For now warn if there is more than one subregion
+    if len(region) > 1:
+        # Raise a warning if the SpectralRegion has more than one subregion, since
+        # the handling for this is perhaps unexpected
+        warnings.warn("A SpectralRegion with multiple subregions was provided as "
+             "input. This may lead to undesired behavior with linear_exciser if "
+             "the subregions overlap.",
+             AstropyUserWarning)
+
+    for subregion in region:
+        # Find the indices of the spectral_axis array corresponding to the subregion
+        wavelengths_in = (wavelengths >= subregion.lower) & (wavelengths < subregion.upper)
+        inclusive_indices = np.nonzero(wavelengths_in)[0]
+        # Now set the flux values for these indices to be a
+        # linear range
+        s, e = max(inclusive_indices[0]-1, 0), min(inclusive_indices[-1]+1,
                                                wavelengths.size-1)
 
-    flux = spectrum.flux.value
-    modified_flux = flux
-    modified_flux[s:e] = np.linspace(flux[s], flux[e], modified_flux[s:e].size)
+        modified_flux[s:e+1] = np.linspace(flux[s], flux[e], modified_flux[s:e+1].size)
+
+        # Add the uncertainty of the two linear interpolation endpoints in
+        # quadrature and apply to the excised region.
+        if new_uncertainty is not None:
+            new_uncertainty[s:e] = np.sqrt(spectrum.uncertainty[s]**2 + spectrum.uncertainty[e]**2)
 
     # Return a new object with the regions excised.
-    return Spectrum1D(flux=modified_flux*spectrum.flux.unit,
+    return Spectrum1D(flux=modified_flux,
                       spectral_axis=wavelengths,
-                      uncertainty=spectrum.uncertainty,
-                      wcs=spectrum.wcs, unit=spectrum.unit,
+                      uncertainty=new_uncertainty,
+                      wcs=spectrum.wcs,
+                      mask = spectrum.mask,
                       velocity_convention=spectrum.velocity_convention,
-                      rest_value=spectrum.rest_value)
+                      rest_value=spectrum.rest_value,
+                      radial_velocity = spectrum.radial_velocity)
 
 
-def excise_regions(spectrum, regions, exciser=linear_exciser):
+def excise_regions(spectrum, regions, exciser=true_exciser):
     """
-    Apply a convolution based smoothing to the spectrum. The kernel must be one
-    of the 1D kernels defined in `astropy.convolution`.
-
-    This method can be used along but also is used by other specific methods below.
+    Method to replace the flux in the defined regions of the spectrum with
+    interpolated values produced by the function given in the ``exciser``
+    argument.
```

Comment:
This isn't particularly true in the case of a `true_exciser`. Perhaps something like

```suggestion
    Method to remove or replace the flux in the defined regions of the spectrum 
    depending on the function provided in the ``exciser`` argument.
```